File: 7KYU/Geometry Basics: Triangle Perimeter in 2D/solution.js
function trianglePerimeter(triangle){
    console.log(triangle.a.x + "," + triangle.a.y);
    console.log(triangle.b.x + "," + triangle.b.y);
    console.log(triangle.c.x + "," + triangle.c.y);
    
    const ab = Math.sqrt(Math.abs(triangle.a.x - triangle.b.x) * Math.abs(triangle.a.x - triangle.b.x)
                       + Math.abs(triangle.a.y - triangle.b.y) * Math.abs(triangle.a.y - triangle.b.y));
    
    const bc = Math.sqrt(Math.abs(triangle.b.x - triangle.c.x) * Math.abs(triangle.b.x - triangle.c.x)
                       + Math.abs(triangle.b.y - triangle.c.y) * Math.abs(triangle.b.y - triangle.c.y));
    
    const ac = Math.sqrt(Math.abs(triangle.a.x - triangle.c.x) * Math.abs(triangle.a.x - triangle.c.x)
                       + Math.abs(triangle.a.y - triangle.c.y) * Math.abs(triangle.a.y - triangle.c.y));
    
    return ab + bc + ac;
  }

File: 7KYU/Geometry Basics: Triangle Perimeter in 2D/solution.c
#include<stdlib.h>
#include <math.h>

typedef struct { double x, y;   } Point;
typedef struct { Point a, b, c; } Triangle;

double triangle_perimeter(Triangle t) {
    double ab = sqrt(fabs(t.a.x - t.b.x) * fabs(t.a.x - t.b.x) + fabs(t.a.y - t.b.y) * fabs(t.a.y - t.b.y));
    double bc = sqrt(fabs(t.b.x - t.c.x) * fabs(t.b.x - t.c.x) + fabs(t.b.y - t.c.y) * fabs(t.b.y - t.c.y));
    double ac = sqrt(fabs(t.a.x - t.c.x) * fabs(t.a.x - t.c.x) + fabs(t.a.y - t.c.y) * fabs(t.a.y - t.c.y));
    return ab + bc + ac;
}